tests/filecheck/dialects/csl/ops.mlir

```diff
@@ -70,6 +70,10 @@ csl.func @initialize() {
     %many_arr_ptr = "csl.addressof"(%arr) : (memref<10xf32>) -> !csl.ptr<f32, #csl<ptr_kind many>, #csl<ptr_const const>>
     %single_arr_ptr = "csl.addressof"(%arr) : (memref<10xf32>) -> !csl.ptr<memref<10xf32>, #csl<ptr_kind single>, #csl<ptr_const const>>
 
+    %dsd_1d = "csl.get_dsd"(%arr, %scalar) : (memref<10xf32>, i32) -> !csl<dsd mem1d_dsd>
+    %dsd_2d = "csl.get_dsd"(%arr, %scalar, %scalar) : (memref<10xf32>, i32, i32) -> !csl<dsd mem4d_dsd>
```

Comment:
Two questions here: 
1. Can we not just read the sizes from the memref type?
2. Can we have two sizes passed to the get_dsd if the underlying array is 1D?

Reply:
1. What if you don't want to iterate over the entire buffer?
2. It will fail to verify.